packages/aws-cdk/test/api/hotswap/lambda-functions-inline-hotswap-deployments.test.ts

```diff
@@ -0,0 +1,110 @@
+import { Lambda } from 'aws-sdk';
+import * as setup from './hotswap-test-setup';
+import { zipString } from '../../../lib/api/hotswap/lambda-functions';
+
+let mockUpdateLambdaCode: (params: Lambda.Types.UpdateFunctionCodeRequest) => Lambda.Types.FunctionConfiguration;
+let mockTagResource: (params: Lambda.Types.TagResourceRequest) => {};
+let mockUntagResource: (params: Lambda.Types.UntagResourceRequest) => {};
+let hotswapMockSdkProvider: setup.HotswapMockSdkProvider;
+
+beforeEach(() => {
+  hotswapMockSdkProvider = setup.setupHotswapTests();
+  mockUpdateLambdaCode = jest.fn();
+  mockTagResource = jest.fn();
+  mockUntagResource = jest.fn();
+  hotswapMockSdkProvider.stubLambda({
+    updateFunctionCode: mockUpdateLambdaCode,
+    tagResource: mockTagResource,
+    untagResource: mockUntagResource,
+  });
+});
+
+test('calls the updateLambdaCode() API when it receives only a code difference in a Lambda function (Inline Node.js code)', async () => {
+  // GIVEN
+  setup.setCurrentCfnStackTemplate({
+    Resources: {
+      Func: {
+        Type: 'AWS::Lambda::Function',
+        Properties: {
+          Code: {
+            ZipFile: 'exports.handler = () => {return true}',
+          },
+          Runtime: 'nodejs14.x',
+          FunctionName: 'my-function',
+        },
+      },
+    },
+  });
+  const newCode = 'exports.handler = () => {return false}';
+  const cdkStackArtifact = setup.cdkStackArtifactOf({
+    template: {
+      Resources: {
+        Func: {
+          Type: 'AWS::Lambda::Function',
+          Properties: {
+            Code: {
+              ZipFile: newCode,
+            },
+            Runtime: 'nodejs14.x',
+            FunctionName: 'my-function',
+          },
+        },
+      },
+    },
+  });
+
+  // WHEN
+  const deployStackResult = await hotswapMockSdkProvider.tryHotswapDeployment(cdkStackArtifact);
+
+  // THEN
+  expect(deployStackResult).not.toBeUndefined();
+  expect(mockUpdateLambdaCode).toHaveBeenCalledWith({
+    FunctionName: 'my-function',
+    ZipFile: await zipString('index.js', newCode),
```

Comment:
I hate exporting things from production code just to use them in tests.

Considering we are not actually verifying here that we ZIP the strings correctly (if I change `zipString()` to `return 'abc';`, these tests would still pass), let's make `zipString()` private (not exported from its file), and instead use a Jest matcher here, `expect.any(String)`.